### ISSUE TITLE
A question about finetuning on RSICD

### ISSUE THREAD

**Issue by mcx-mcx** [om-ai-lab/RS5M#12]
Thank you very much for your outstanding work！
I have a question that I haven't quite understood. When fine-tuning your RS5M model on RSICD or RSITMD using the methods outlined in the paper (infoNCE, lr=1e-6), I did not achieve the expected performance. Taking RSICD as an example, the paper and the weights you provided for RS5M RET-2 version result in an accuracy around 38, but when I fine-tuned using my own RS5M VitB32 version, the result was around 34. Could you provide more details on fine-tuning RET-2 or RSICD so that I can better replicate the process? Thank you very much.

**Comment by zilunzhang** [comment]
> Thank you very much for your outstanding work！ I have a question that I haven't quite understood. When fine-tuning your RS5M model on RSICD or RSITMD using the methods outlined in the paper (infoNCE, lr=1e-6), I did not achieve the expected performance. Taking RSICD as an example, the paper and the weights you provided for RS5M RET-2 version result in an accuracy around 38, but when I fine-tuned using my own RS5M VitB32 version, the result was around 34. Could you provide more details on fine-tuning RET-2 or RSICD so that I can better replicate the process? Thank you very much.

Hi

We used the ITRA codebase to tune the model.

The RET-2 data was made by: https://github.com/ChenDelong1999/RemoteCLIP/issues/13#issuecomment-1836218269

Hyperparams for RET-2 model:

--lr: 5e-06
--weight_decay: 0.5
--batch_size: 600
--epochs: 7
--warmup 100
--max_grad_norm: 50.0

Best,

Zilun

**Comment by mcx-mcx** [comment]
Thank you very much!

**Comment by lemyx** [comment]
Thanks for your great work!

I wanna reproduce the following results, could you share the hyperparameters related to lr, weight_decay, bs, epochs, warmup and max_grad_norm?

<img width="1216" alt="image" src="https://github.com/om-ai-lab/RS5M/assets/119666897/3da7e753-e8a5-4349-a263-c6829127ebe3">
<img width="1216" alt="image" src="https://github.com/om-ai-lab/RS5M/assets/119666897/1cb47ca7-1ceb-43b6-84ca-28440153bf10">

**Comment by zilunzhang** [comment]
> Thanks for your great work!
> 
> I wanna reproduce the following results, could you share the hyperparameters related to lr, weight_decay, bs, epochs, warmup and max_grad_norm?
> 
> <img alt="image" width="1216" src="https://private-user-images.githubusercontent.com/119666897/306146002-3da7e753-e8a5-4349-a263-c6829127ebe3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc2MTY1NTksIm5iZiI6MTcyNzYxNjI1OSwicGF0aCI6Ii8xMTk2NjY4OTcvMzA2MTQ2MDAyLTNkYTdlNzUzLWU4YTUtNDM0OS1hMjYzLWM2ODI5MTI3ZWJlMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyOVQxMzI0MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMjExYjY2N2JkYTkwZGI3NTkxZDU3NzU3ZDZjZjE5ZTAxM2ZjYmY2MTExZmRhMzg3MDNlY2JmZGNiMTYwMzNjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9._jJ_A5rbagDnxHGahMh9g_YPl3kpNXvep4QBd_bJCXQ"> <img alt="image" width="1216" src="https://private-user-images.githubusercontent.com/119666897/306146094-1cb47ca7-1ceb-43b6-84ca-28440153bf10.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc2MTY1NTksIm5iZiI6MTcyNzYxNjI1OSwicGF0aCI6Ii8xMTk2NjY4OTcvMzA2MTQ2MDk0LTFjYjQ3Y2E3LTFjZWItNDNiNi04NGNhLTI4NDQwMTUzYmYxMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyOVQxMzI0MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMjcyMDVkZmY0Y2MyM2JmZDliMDEyMzQ1OGE4YjVjN2YxYjQzZDQ0YWNiYWIyNjRjNDExYWUwM2Y5YWNkMTY4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ULp6l63OXgwYzzQpGx5qFCQiZpEmDOtiFtXBRJ0fIDA">

Sure. Could you leave us an email address?

**Comment by r-deo** [comment]
@zilunzhang I was trying the reproduce FT on RS5M.  But, I was not able to reproduce. Would it be possible to share the hyper parameter and training related modifications? 

Thanks
